test/integration/vpc_e2e_test.go

```diff
@@ -0,0 +1,166 @@
+			// Find if vpc with expected Id is present in the vpc inventory.
+			cmd := fmt.Sprintf("get vpc %v -n %v -o json -o=jsonpath={.spec.Id}", vpcObjectName, ns.Name)
+			out, err := kubeCtl.Cmd(cmd)
+			logf.Log.V(1).Info("Got VPC", "vpcId", out)
+			if err != nil {
+				logf.Log.V(1).Info("Failed to get vpc", "vpc object name", vpcObjectName, "err", err)
+				return false, nil
+			}
+			Expect(out).To(Equal(vpcID))
+			return true, nil
+		})
+		return err
+	}
+
+	verifyNoVmInventory := func() error {
+		err := wait.Poll(time.Second*5, time.Second*30, func() (bool, error) {
+			// Find if any vm is imported from cloud.
+			cmd := fmt.Sprintf("get vm -n %v -o json -o json -o=jsonpath={.items}", ns.Name)
+			out, err := kubeCtl.Cmd(cmd)
+			if err != nil {
+				logf.Log.V(1).Info("Failed to get vms", "namespace", ns.Name, "err", err)
+				return false, nil
+			}
+			Expect(out).To(ContainSubstring("[]"))
+			return true, nil
+		})
+		return err
+	}
+
+	It("Basic Test for VPC", func() {
+		kind := reflect.TypeOf(v1alpha1.VirtualMachine{}).Name()
+		// Create a Namespace
+		if len(ns.Status.Phase) == 0 {
+			// Create ns if not already present.
+			ns.Name = "test-cloud-e2e-" + fmt.Sprintf("%x", rand.Int())
+			ns.Labels = map[string]string{"name": ns.Name}
+			err := k8sClient.Create(context.TODO(), ns)
+			Expect(err).ToNot(HaveOccurred())
+			logf.Log.V(1).Info("Created namespace", "ns", ns.Name)
+		}
+
+		By("Configure CPA and verify VPC inventory")
+		logf.Log.V(1).Info("Configure and create CPA")
+		accountParams := cloudVPC.GetCloudAccountParameters("test-cloud-account"+ns.Name, ns.Name, cloudCluster)
+		err := utils.AddCloudAccount(kubeCtl, accountParams)
+		Expect(err).ToNot(HaveOccurred())
+
+		// Generate kubernetes object name for vpc object.
+		vpcObjectName := ""
+		switch cloudProvider {
+		case string(v1alpha1.AWSCloudProvider):
+			vpcObjectName = vpcID
+		case string(v1alpha1.AzureCloudProvider):
+			vpcObjectName = cloudUtils.GenerateShortResourceIdentifier(vpcID, vpcName)
+		default:
+			logf.Log.Error(nil, "invalid provider", "Provider", cloudProvider)
+		}
+
+		logf.Log.V(1).Info("Get vpcID from VPC inventory", "vpcID", vpcID)
+		err = verifyVpcInventory(vpcObjectName, vpcID)
+		Expect(err).ToNot(HaveOccurred(), "timeout waiting to retrieve vpc")
+		logf.Log.V(1).Info("Verify VM inventory is not imported", "namespace", ns.Name)
+		err = verifyNoVmInventory()
+		Expect(err).ToNot(HaveOccurred(), "timeout waiting to get vm list")
+
+		By("Configure CES and verify VM inventory")
+		entityParams := cloudVPC.GetEntitySelectorParameters("test-entity-selector"+ns.Name, ns.Name, kind)
+		logf.Log.V(1).Info("Create CES and verify VM inventory is imported", "ces", entityParams.Name)
+		err = utils.ConfigureEntitySelectorAndWait(kubeCtl, k8sClient, entityParams, kind, len(vmIDs), ns.Name, false)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Delete CES and verify cloud inventory")
```

Comment:
Organize the tests as
1) Add/Delete CPA 
2) Add/Update/Delete CES
3) Restart nephe. 
have separate tests. In this way, its easier to know what functionality is broken

Reply:
Yes added